SDD 0021 - Cluster Catalog Compilation (#62)
projectsyn/documentation: docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc

It shouldn't be possible for example for a tenant to trigger compilation for clusters of other tenants.

=== Compilation Engine

To compile the catalog for a cluster, a compilation job is created.
The compilation job is a https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion[Kubernetes job] resource which executes the xref:0006-commodore-configuration-catalog-compilation.adoc[catalog compilation engine].
The compilation engine starts a compilation job if the `cluster.syn.tools/compilation-id` annotation has changed on a cluster object.

> By default the compilation engine creates the compile jobs on the respective target clusters (for example via Steward).

Reviewer: ```suggestion
Following a decentralized approach, the compilation engine creates the compile jobs on the respective target clusters (for example via Steward).
```

> To distribute the load the compilation generates and to follow a decentralized approach, these jobs are run on the same cluster as they compile the catalog for.

Reviewer: That sentences is missing something. Also the decentralization is somewhat redundant with the next sentence.

> With the goal to distribute the load the compilation generates and to follow a decentralized approach.

Reviewer: ```suggestion
This removes the need for a central infrastructure which would do nothing most of the time.
```
